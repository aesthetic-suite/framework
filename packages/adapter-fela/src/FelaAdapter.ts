import {
  Adapter,
  ClassName,
  Ruleset,
  Sheet,
  StyleName,
  GLOBAL_STYLE_NAME,
  FontFace,
} from 'aesthetic';
import { getStyleElements, purgeStyles, toArray } from 'aesthetic-utils';
import { combineRules, IRenderer } from 'fela';
import { render } from 'fela-dom';
import { NativeBlock, ParsedBlock } from './types';

export default class FelaAdapter extends Adapter<NativeBlock, ParsedBlock> {
  fela: IRenderer;

  keyframes: { [animationName: string]: ClassName } = {};

  constructor(fela: IRenderer) {
    super();

    this.fela = fela;

    this.syntax
      .on('attribute', this.handleNested)
      .on('css', this.handleCss)
      .on('fallback', this.handleFallback)
      .on('font-face', this.handleFontFace)
      .on('global', this.handleGlobal)
      .on('keyframe', this.handleKeyframe)
      .on('media', this.handleMedia)
      .on('property', this.handleProperty)
      .on('pseudo', this.handleNested)
      .on('selector', this.handleNested)
      .on('support', this.handleSupport);
  }

  flushStyles() {
    render(this.fela);
  }

  transformToClassName(styles: ParsedBlock[]): ClassName {
    return this.fela.renderRule(combineRules(...styles.map(style => () => style)), {});
  }

  purgeStyles(styleName?: StyleName) {
    purgeStyles(getStyleElements('data-fela-type'), styleName === GLOBAL_STYLE_NAME);
  }

  // http://fela.js.org/docs/api/fela/Renderer.html
  private handleCss = (css: string) => {
    this.fela.renderStatic(css);
  };

  // https://github.com/rofrischmann/fela/tree/master/packages/fela-plugin-fallback-value
  private handleFallback = (
    ruleset: Ruleset<NativeBlock>,
    name: keyof NativeBlock,
    value: unknown[],
  ) => {
    ruleset.addProperty(name, [...value, ...toArray(ruleset.properties[name])]);
  };

  // http://fela.js.org/docs/basics/Fonts.html
  // http://fela.js.org/docs/api/fela/Renderer.html
  private handleFontFace = (
    sheet: Sheet<NativeBlock>,
    fontFaces: Ruleset<NativeBlock>[],
    fontFamily: string,
    srcPaths: string[][],
  ) => {
    fontFaces.forEach((face, i) => {
      const { local, ...style } = face.toObject() as FontFace;

      this.fela.renderFont(fontFamily, srcPaths[i], {
        ...style,
        localAlias: local,
      });
    });
  };

  // http://fela.js.org/docs/advanced/StaticStyle.html
  // http://fela.js.org/docs/api/fela/Renderer.html#renderstaticstyle-selector
  private handleGlobal = (
    sheet: Sheet<NativeBlock>,
    selector: string,
    ruleset: Ruleset<NativeBlock>,
  ) => {
    this.fela.renderStatic(ruleset.toObject(), selector);
  };

  // http://fela.js.org/docs/basics/Keyframes.html
  // http://fela.js.org/docs/basics/Renderer.html#renderkeyframe
  private handleKeyframe = (
    sheet: Sheet<NativeBlock>,
    keyframe: Ruleset<NativeBlock>,
    animationName: string,
  ) => {
    this.keyframes[animationName] = this.fela.renderKeyframe(
      // @ts-ignore Annoying to type
      () => keyframe.toObject(),
      {},
    );
  };

  // http://fela.js.org/docs/basics/Rules.html#3-media-queries
  private handleMedia = (
    ruleset: Ruleset<NativeBlock>,
    query: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(`@media ${query}`, value);
  };

  // http://fela.js.org/docs/basics/Rules.html#style-object
  private handleNested = (
    ruleset: Ruleset<NativeBlock>,
    selector: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(selector, value);
  };

  // http://fela.js.org/docs/basics/Rules.html
  private handleProperty = (
    ruleset: Ruleset<NativeBlock>,
    name: keyof NativeBlock,
    value: unknown,
  ) => {
    if (name === 'animationName') {
      ruleset.addProperty(
        name,
        this.syntax.injectKeyframes(String(value), this.keyframes).join(', '),
      );
    } else {
      ruleset.addProperty(name, value);
    }
  };

  // http://fela.js.org/docs/basics/Rules.html
  private handleSupport = (
    ruleset: Ruleset<NativeBlock>,
    query: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(`@supports ${query}`, value);
  };
}
