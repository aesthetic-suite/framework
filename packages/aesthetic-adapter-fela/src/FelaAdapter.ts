/**
 * @copyright   2017, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import { Adapter, ClassName, UnifiedSyntax, FontFace, Keyframes, injectKeyframes } from 'aesthetic';
import { createRenderer, combineRules, IRenderer } from 'fela';
import { render } from 'fela-dom';
import { StyleSheet, Declaration } from './types';

export default class FelaAdapter extends Adapter<StyleSheet, Declaration> {
  fela: IRenderer;

  keyframesCache: { [animationName: string]: string } = {};

  constructor(fela?: IRenderer) {
    super();

    this.fela = fela || createRenderer();

    render(this.fela);
  }

  transform(...styles: Declaration[]): ClassName {
    return this.fela.renderRule(
      combineRules(...styles.map(style => (typeof style === 'function' ? style : () => style))),
      {},
    );
  }

  unify(syntax: UnifiedSyntax<StyleSheet, Declaration>) {
    syntax
      .on('property', this.handleProperty)
      .on('@charset', syntax.createUnsupportedHandler('@charset'))
      .on('@font-face', this.handleFontFace)
      .on('@global', this.handleGlobal)
      .on('@import', syntax.createUnsupportedHandler('@import'))
      .on('@keyframes', this.handleKeyframe)
      .on('@namespace', syntax.createUnsupportedHandler('@namespace'))
      .on('@page', syntax.createUnsupportedHandler('@page'))
      .on('@viewport', syntax.createUnsupportedHandler('@viewport'));
  }

  // http://fela.js.org/docs/basics/Fonts.html
  // http://fela.js.org/docs/basics/Renderer.html#renderfont
  handleFontFace = (
    styleSheet: StyleSheet,
    fontFaces: FontFace[],
    fontFamily: string,
    srcPaths: string[][],
  ) => {
    fontFaces.map((face, i) => this.fela.renderFont(fontFamily, srcPaths[i], face));
  };

  // http://fela.js.org/docs/advanced/StaticStyle.html
  // http://fela.js.org/docs/api/fela/Renderer.html#renderstaticstyle-selector
  handleGlobal = (styleSheet: StyleSheet, declaration: Declaration, selector: string) => {
    this.fela.renderStatic(declaration, selector);
  };

  // http://fela.js.org/docs/basics/Keyframes.html
  // http://fela.js.org/docs/basics/Renderer.html#renderkeyframe
  handleKeyframe = (styleSheet: StyleSheet, keyframes: Keyframes, animationName: string) => {
    this.keyframesCache[animationName] = this.fela.renderKeyframe(() => keyframes);
  };

  handleProperty = (declaration: Declaration, value: any, property: string) => {
    if (property === 'animationName') {
      declaration[property] = injectKeyframes(value, this.keyframesCache).join(', ');
    } else {
      declaration[property] = style;
    }
  };
}
