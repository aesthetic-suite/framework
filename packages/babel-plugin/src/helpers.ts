import { types as t } from '@babel/core';
import { createDebugger } from '@boost/debug';

export const debug = createDebugger('aesthetic:babel');

export function isFunction(node: t.Node) {
  return t.isFunctionExpression(node) || t.isArrowFunctionExpression(node);
}
