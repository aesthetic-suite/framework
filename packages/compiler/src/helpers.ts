import { ColorShade } from '@aesthetic/system';

export function formatShade(value: number | string): ColorShade {
	return String(value).padStart(2, '0').slice(0, 2) as ColorShade;
}

export function formatUnit(value: number): string {
	return value.toFixed(2).replace('.00', '');
}
