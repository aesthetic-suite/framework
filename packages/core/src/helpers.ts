import { Theme } from '@aesthetic/system';
import { objectLoop } from '@aesthetic/utils';
import { SheetParams } from './types';

export function createCacheKey(params: SheetParams, type: string): string {
  let key = type;

  // Since all other values are scalars, we can just join the values.
  // This is 3x faster than JSON.stringify(), and 1.5x faster than Object.values()!
  objectLoop(params, (value) => {
    key += value;
  });

  return key;
}

export function createDefaultParams(theme: Theme<{}>, params: SheetParams): Required<SheetParams> {
  return {
    contrast: theme.contrast,
    deterministic: false,
    direction: 'ltr',
    scheme: theme.scheme,
    theme: theme.name,
    unit: 'px',
    vendor: false,
    ...params,
  };
}
