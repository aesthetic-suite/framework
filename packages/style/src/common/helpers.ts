/* eslint-disable no-magic-numbers */

import { Value, MaybeValue, CSS, Sheet } from '@aesthetic/types';
import { hyphenate } from '@aesthetic/utils';
import { IMPORT_RULE, STYLE_RULE } from './constants';

export function insertRule(sheet: Sheet, rule: CSS, index?: number): number {
  try {
    return sheet.insertRule(rule, index ?? sheet.cssRules.length);
  } catch {
    // Vendor prefixed properties, pseudos, etc, that are inserted
    // into different vendors will trigger a failure. For example,
    // `-moz` or `-ms` being inserted into WebKit.
    // There's no easy way around this, so let's just ignore the
    // error so that subsequent styles are inserted.
    // istanbul ignore next
    return -1;
  }
}

export function insertAtRule(sheet: Sheet, rule: CSS): number {
  const { length } = sheet.cssRules;
  let index = 0;

  // At-rules must be inserted before normal style rules.
  for (let i = 0; i <= length; i += 1) {
    index = i;

    if (sheet.cssRules[i]?.type === STYLE_RULE) {
      break;
    }
  }

  return insertRule(sheet, rule, index);
}

export function insertImportRule(sheet: Sheet, rule: CSS): number {
  const { length } = sheet.cssRules;
  let index = 0;

  // Import rules must be inserted at the top of the style sheet,
  // but we also want to persist the existing order.
  for (let i = 0; i <= length; i += 1) {
    index = i;

    if (sheet.cssRules[i]?.type !== IMPORT_RULE) {
      break;
    }
  }

  return insertRule(sheet, rule, index);
}

export function isAtRule(value: string): boolean {
  return value[0] === '@';
}

export function isImportRule(value: string): boolean {
  return value.slice(0, 7) === '@import';
}

export function isMediaRule(value: string): boolean {
  return value.slice(0, 6) === '@media';
}

export function isNestedSelector(value: string): boolean {
  const char = value[0];

  return (
    char === ':' ||
    char === '[' ||
    char === '>' ||
    char === '~' ||
    char === '+' ||
    char === '*' ||
    char === '|'
  );
}

export function isSupportsRule(value: string): boolean {
  return value.slice(0, 9) === '@supports';
}

const unitlessProperties = new Set<string>();

[
  'animationIterationCount',
  'borderImage',
  'borderImageOutset',
  'borderImageSlice',
  'borderImageWidth',
  'columnCount',
  'columns',
  'flex',
  'flexGrow',
  'flexPositive',
  'flexShrink',
  'flexNegative',
  'flexOrder',
  'fontWeight',
  'gridArea',
  'gridRow',
  'gridRowEnd',
  'gridRowSpan',
  'gridRowStart',
  'gridColumn',
  'gridColumnEnd',
  'gridColumnSpan',
  'gridColumnStart',
  'lineClamp',
  'lineHeight',
  'maskBorder',
  'maskBorderOutset',
  'maskBorderSlice',
  'maskBorderWidth',
  'opacity',
  'order',
  'orphans',
  'tabSize',
  'widows',
  'zIndex',
  'zoom',
  // SVG
  'fillOpacity',
  'floodOpacity',
  'stopOpacity',
  'strokeDasharray',
  'strokeDashoffset',
  'strokeMiterlimit',
  'strokeOpacity',
  'strokeWidth',
].forEach((property) => {
  unitlessProperties.add(property);
  unitlessProperties.add(hyphenate(property));
});

export function isUnitlessProperty(property: string): boolean {
  return unitlessProperties.has(property);
}

export function isValidValue(property: string, value: MaybeValue | boolean): value is Value {
  if (value === null || value === undefined || value === true || value === false || value === '') {
    if (__DEV__) {
      // eslint-disable-next-line no-console
      console.warn(`Invalid value "${value}" for "${property}".`);
    }

    return false;
  }

  return true;
}

export function isVariable(value: string): boolean {
  return value.slice(0, 2) === '--';
}

export function joinQueries(prev: string | undefined, next: string): string {
  if (prev) {
    return `${prev} and ${next}`;
  }

  return next;
}
