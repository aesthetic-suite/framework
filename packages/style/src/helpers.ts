/* eslint-disable no-magic-numbers */

import { Value, MaybeValue } from '@aesthetic/types';
import { hyphenate } from '@aesthetic/utils';

export function isAtRule(value: string): boolean {
  return value[0] === '@';
}

export function isImportRule(value: string): boolean {
  return value.slice(0, 7) === '@import' && value[7] === ' ';
}

export function isNestedSelector(value: string): boolean {
  const char = value[0];

  return (
    char === ':' ||
    char === '[' ||
    char === '>' ||
    char === '~' ||
    char === '+' ||
    char === '*' ||
    char === '|'
  );
}

const unitlessProperties = new Set<string>();

[
  'animationIterationCount',
  'borderImage',
  'borderImageOutset',
  'borderImageSlice',
  'borderImageWidth',
  'columnCount',
  'columns',
  'flex',
  'flexGrow',
  'flexPositive',
  'flexShrink',
  'flexNegative',
  'flexOrder',
  'fontWeight',
  'gridArea',
  'gridRow',
  'gridRowEnd',
  'gridRowSpan',
  'gridRowStart',
  'gridColumn',
  'gridColumnEnd',
  'gridColumnSpan',
  'gridColumnStart',
  'lineClamp',
  'lineHeight',
  'maskBorder',
  'maskBorderOutset',
  'maskBorderSlice',
  'maskBorderWidth',
  'opacity',
  'order',
  'orphans',
  'tabSize',
  'widows',
  'zIndex',
  'zoom',
  // SVG
  'fillOpacity',
  'floodOpacity',
  'stopOpacity',
  'strokeDasharray',
  'strokeDashoffset',
  'strokeMiterlimit',
  'strokeOpacity',
  'strokeWidth',
].forEach((property) => {
  unitlessProperties.add(property);
  unitlessProperties.add(hyphenate(property));
});

export function isUnitlessProperty(property: string): boolean {
  return unitlessProperties.has(property);
}

export function isValidValue(property: string, value: MaybeValue | boolean): value is Value {
  if (value === null || value === undefined || value === true || value === false || value === '') {
    if (__DEV__) {
      console.warn(`Invalid value "${value}" for "${property}".`);
    }

    return false;
  }

  return true;
}

export function isVariable(value: string): boolean {
  return value.slice(0, 2) === '--';
}
