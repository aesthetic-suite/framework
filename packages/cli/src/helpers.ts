import { CONFIG_FOLDER } from '@aesthetic/compiler';
import { Path } from '@boost/common';
import { CWD } from './constants';

export function getConfigFolderDir(name: string): Path {
  return new Path(CWD, CONFIG_FOLDER, name);
}

export function validateSystemName(value: string) {
  if (value && !value.match(/^[-a-z0-9]+$/giu)) {
    throw new Error('Name may only contain numbers, letters, and dashes.');
  }
}
