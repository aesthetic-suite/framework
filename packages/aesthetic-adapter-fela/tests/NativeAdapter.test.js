/* eslint-disable sort-keys */

import { createRenderer } from 'fela';
import webPreset from 'fela-preset-web';
import FelaAdapter from '../src/NativeAdapter';
import {
  FONT_ROBOTO_FLAT_SRC,
  FONT_CIRCULAR_MULTIPLE,
  FONT_CIRCULAR_MULTIPLE_FLAT_SRC,
  KEYFRAME_FADE,
  SYNTAX_NATIVE_PARTIAL,
  SYNTAX_ATTRIBUTE,
  SYNTAX_DESCENDANT,
  SYNTAX_PSEUDO,
} from '../../../tests/mocks';
import { renderFelaStyles } from '../../../tests/helpers';

describe('aesthetic-adapter-fela/NativeAdapter', () => {
  let instance;

  beforeEach(() => {
    instance = new FelaAdapter(
      createRenderer({
        plugins: [...webPreset],
      }),
    );
  });

  it('can customize the fela instance through the constructor', () => {
    const renderer = createRenderer({ plugins: [] });
    const newInstance = new FelaAdapter(renderer);

    expect(newInstance.fela).not.toEqual(instance.fela);
  });

  it('can transform dynamic styles', () => {
    expect(
      instance.transform({
        width: 10,
        height: 10,
      }),
    ).toBe('a b');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('transforms style declarations into class names', () => {
    expect(instance.transform(instance.create(SYNTAX_NATIVE_PARTIAL).button)).toBe(
      'a b c d e f g h i j k l m n o p q r s t u v w',
    );
  });

  it('combines different style declarations into unique class names', () => {
    const sheet = instance.create({
      foo: {
        color: 'red',
        display: 'block',
      },
      bar: {
        color: 'green',
        margin: 5,
      },
      baz: {
        color: 'blue',
        padding: 5,
      },
    });

    expect(instance.transform(sheet.foo)).toBe('a b');
    expect(instance.transform(sheet.bar)).toBe('c d');
    expect(instance.transform(sheet.baz)).toBe('e f');
    expect(instance.transform(sheet.foo, sheet.baz)).toBe('e b f');
    expect(instance.transform(sheet.bar, sheet.foo)).toBe('a d b');
  });

  it('handles attribute selectors', () => {
    expect(instance.transform(instance.create(SYNTAX_ATTRIBUTE).attr)).toBe('a b');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('handles descendant selectors', () => {
    expect(instance.transform(instance.create(SYNTAX_DESCENDANT).list)).toBe('a b c');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('handles pseudo selectors', () => {
    expect(instance.transform(instance.create(SYNTAX_PSEUDO).pseudo)).toBe('a b c');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('handles fallbacks', () => {
    expect(
      instance.transform(
        instance.create({
          fallback: {
            background: ['red', 'linear-gradient(...)'],
            display: ['box', 'flex-box', 'flex'],
          },
        }).fallback,
      ),
    ).toBe('a b');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('handles font faces', () => {
    instance.fela.renderFont('Roboto', ['fonts/roboto.woff2'], FONT_ROBOTO_FLAT_SRC);

    expect(
      instance.transform(
        instance.create({
          font: {
            fontFamily: 'Roboto',
            fontSize: 20,
          },
        }).font,
      ),
    ).toBe('a b');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('handles multiple font faces', () => {
    FONT_CIRCULAR_MULTIPLE_FLAT_SRC.forEach((font, i) => {
      instance.fela.renderFont('Circular', FONT_CIRCULAR_MULTIPLE[i].srcPaths, font);
    });

    expect(
      instance.transform(
        instance.create({
          font: {
            fontFamily: 'Circular',
            fontSize: 20,
          },
        }).font,
      ),
    ).toBe('a b');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('handles animations', () => {
    expect(
      instance.transform(
        instance.create({
          animation: {
            animationName: instance.fela.renderKeyframe(() => KEYFRAME_FADE),
            animationDuration: '3s, 1200ms',
            animationIterationCount: 'infinite',
          },
        }).animation,
      ),
    ).toBe('a b c');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('handles media queries', () => {
    expect(
      instance.transform(
        instance.create({
          media: {
            color: 'red',
            '@media (min-width: 300px)': {
              color: 'blue',
            },
            '@media (max-width: 1000px)': {
              color: 'green',
            },
          },
        }).media,
      ),
    ).toBe('a b c');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });

  it('handles supports', () => {
    expect(
      instance.transform(
        instance.create({
          sup: {
            display: 'block',
            '@supports (display: flex)': {
              display: 'flex',
            },
            '@supports not (display: flex)': {
              float: 'left',
            },
          },
        }).sup,
      ),
    ).toBe('a b c');

    expect(renderFelaStyles(instance)).toMatchSnapshot();
  });
});
