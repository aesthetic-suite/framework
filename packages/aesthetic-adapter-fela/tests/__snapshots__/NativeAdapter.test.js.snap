// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aesthetic-adapter-fela/NativeAdapter can transform dynamic styles 1`] = `".a{width:10px}.b{height:10px}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles animations 1`] = `"@-webkit-keyframes k1{from{opacity:0}to{opacity:1}}@-moz-keyframes k1{from{opacity:0}to{opacity:1}}@keyframes k1{from{opacity:0}to{opacity:1}}.a{-webkit-animation-name:k1;animation-name:k1}.b{-webkit-animation-duration:3s, 1200ms;animation-duration:3s, 1200ms}.c{-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles attribute selectors 1`] = `".a{display:block}.b[disabled]{opacity:0.5}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles descendant selectors 1`] = `".a{margin:0px}.b{padding:0px}.c> li{list-style:bullet}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles fallbacks 1`] = `".a{background:red;background:-webkit-linear-gradient(...);background:-moz-linear-gradient(...);background:linear-gradient(...)}.b{display:box;display:flex-box;display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles font faces 1`] = `"@font-face{font-family:\\"Roboto\\";font-style:normal;font-weight:normal;src:url('fonts/roboto.woff2') format('woff2')}.a{font-family:Roboto}.b{font-size:20px}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles media queries 1`] = `".a{color:red}@media (min-width: 300px){.b{color:blue}}@media (max-width: 1000px){.c{color:green}}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles multiple font faces 1`] = `"@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:normal;src:url('fonts/Circular.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:italic;font-weight:normal;src:url('fonts/Circular-Italic.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:300;src:url('fonts/Circular-Light.woff2') format('woff2')}@font-face{font-family:\\"Circular\\";font-style:normal;font-weight:700;src:url('fonts/Circular-Bold.woff2') format('woff2')}.a{font-family:Circular}.b{font-size:20px}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles pseudo selectors 1`] = `".a{position:fixed}.b::before{position:absolute}.c:hover{position:static}"`;

exports[`aesthetic-adapter-fela/NativeAdapter handles supports 1`] = `".a{display:block}@supports (display: flex){.b{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}}@supports not (display: flex){.c{float:left}}"`;
